doc/api/esm.md

```diff
@@ -1096,12 +1096,132 @@ export async function resolve(specifier,
 With this loader, running:
 
 ```console
-NODE_OPTIONS='--experimental-loader ./custom-loader.mjs' node x.js
+NODE_OPTIONS='--experimental-loader ./custom-loader.mjs' node main.js
 ```
 
-would load the module `x.js` as an ES module with relative resolution support
+would load the module `main.js` as an ES module with relative resolution support
 (with `node_modules` loading skipped in this example).
 
+### Get source hook
+
+The `getSource` hook provides a way to define a custom method for retrieving
+the source code of an ES module specifier. This would allow a loader to
+potentially avoid reading files from disk.
+
+```js
+import { get } from 'https';
+
+/**
+ * @param {object} module
+ * @param {string} module.url
+ * @param {string} module.format
+ * @param {function} defaultGetSource
+ * @returns {string|Promise}
+ */
+export async function getSource({ url, format }, defaultGetSource) {
```

Comment:
this should not include a `format` argument as not all situations can provide that ahead of time such as HTTP. it should return a format field in its return value as well. Also, to support WASM and to align with the current implementation, it should be returning a buffer not a string (some formats do coerce to string though/we could document that).